crates/bindings-csharp/Runtime/Internal/Module.cs

```diff
@@ -248,26 +249,34 @@ public static Buffer __describe_module__()
 
     public static Buffer __call_reducer__(
         uint id,
-        Buffer caller_identity,
-        Buffer caller_address,
+        ulong sender_0,
+        ulong sender_1,
+        ulong sender_2,
+        ulong sender_3,
+        ulong address_0,
+        ulong address_1,
         DateTimeOffsetRepr timestamp,
         Buffer args
     )
     {
+        // Piece together the sender identity.
+        ulong[] sender_arr = { sender_0, sender_1, sender_2, sender_3 };
+        byte[] sender_bytes = new byte[sender_arr.Length * sizeof(ulong)];
+        System.Buffer.BlockCopy(sender_arr, 0, sender_bytes, 0, sender_bytes.Length);
+        var sender = Identity.From(sender_bytes);
+
+        // Piece together the sender address.
+        ulong[] addr_arr = { address_0, address_1 };
+        byte[] addr_bytes = new byte[addr_arr.Length * sizeof(ulong)];
+        System.Buffer.BlockCopy(addr_arr, 0, addr_bytes, 0, addr_bytes.Length);
+        var address = Address.From(addr_bytes);
```

Comment:
Slightly simpler:

```suggestion
        // Piece together the sender identity.
        var sender = Identity.From(MemoryMarshal.AsBytes([sender_0, sender_1, sender_2, sender_3]).ToArray());

        // Piece together the sender address.
        var address = Address.From(MemoryMarshal.AsBytes([address_0, address_1]).ToArray());
```

Reply:
(feel free to split back `sender_bytes` and `address_bytes` if it's more readable that way)